test/terminal.js

```diff
@@ -0,0 +1,218 @@
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('home');
+      terminal.cd('home');
+      terminal.cd('../');
+      terminal.pwd();
+
+      expect(console.log.calledWith('/root')).to.be.true;
+    });
+
+    it('Should return "/root" as the current directory after going to home and going back to root using "home/.." in one step', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('home');
+      terminal.cd('home/..');
+      terminal.pwd();
+
+      expect(console.log.calledWith('/root')).to.be.true;
+    });
+
+    it('Should return "/root" as the current directory after going to guille and going back to root using "../.." in one step', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('home/guille');
+      terminal.cd('home/guille');
+      terminal.cd('../../');
+      terminal.pwd();
+
+      expect(console.log.calledWith('/root')).to.be.true;
+    });
+
+    it('Should return "File or directory does not exist" when trying to go to an unexistent foo directory', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.cd('foo');
+
+      expect(console.log.calledWith('File or directory does not exist')).to.be.true;
+    });
+  });
+
+  describe('Terminal mkdir: create directory', function() {
+    it('Should create a new "foo" directory in the root', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('foo');
+      terminal.ls();
+
+      expect(console.log.calledWith('foo')).to.be.true;
+    });
+
+    it('Should print "File or directory does not exist" when trying to create a new directory in an unexistent directory', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('unexistent/bar');
+
+      expect(console.log.calledWith('File or directory does not exist')).to.be.true;
+    });
+
+    it('Should print "File already exists" when trying to create a directory that already exists', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('foo');
+      terminal.mkdir('foo');
+
+      expect(console.log.calledWith('File already exists')).to.be.true;
+    });
+  });
+
+  describe('Terminal touch: create regular file', function() {
+    it('Should create a new "foo.txt" file in the root', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.touch('foo.txt');
+      terminal.ls();
+
+      expect(console.log.calledWith('foo.txt')).to.be.true;
```

Comment:
This expect is dificult to read, Instead use something like: 
```
console.log.should.have.been.calledOnceWith('foo.txt');
```
Try to form a sentence with your assertion